test/test_spectral_ops.py

```diff
@@ -375,21 +404,25 @@ def fn(t: torch.Tensor, s: Optional[List[int]], dim: List[int] = (-2, -1), norm:
 
                 torch_fns = (torch_fn, torch.jit.script(fn))
 
+                if torch.version.hip is not None:
+                    valid_input = self._generate_valid_rocfft_input(input, torch_fn)
+                else:
+                    valid_input = input
+
                 # Once with dim defaulted
-                input_np = input.cpu().numpy()
+                input_np = valid_input.cpu().numpy()
                 expected = numpy_fn(input_np, s, norm=norm)
                 for fn in torch_fns:
-                    actual = fn(input, s, norm=norm)
+                    actual = fn(valid_input, s, norm=norm)
                     self.assertEqual(actual, expected)
 
                 # Once with explicit dims
                 dim = (1, 0)
-                expected = numpy_fn(input.cpu(), s, dim, norm)
+                expected = numpy_fn(valid_input.cpu(), s, dim, norm)
                 for fn in torch_fns:
-                    actual = fn(input, s, dim, norm)
+                    actual = fn(valid_input, s, dim, norm)
```

Comment:
Why is the pattern here different from the others?  In the other places you use
```
            if torch.version.hip is not None:
                input = self._generate_valid_rocfft_input(input, op)
```
But here you are creating a new `valid_input` variable instead of overwriting `input`.  Was this intentional?

Reply:
Yes, in this case, I need to use a temp variable (valid_input) otherwise It overwrites the input variable in line 397 which screws up the unit test.

Reply:
Thanks for confirming.  LGTM then.